lib/afmotion/http_client.rb

```diff
@@ -62,6 +65,12 @@ def request_serializer(serializer)
       else
         @operation_manager.requestSerializer = serializer
       end
+      reinit_options
+    end
+
+    def reinit_options
+      @headers.each{ |h,v| header(h, v) } if !@headers.nil? && !@headers.empty?
```

Comment:
One more thing: Calling `header(h, v)` modifies the `@headers` object. Although this doesn't have any weird effects right now, it's laying a trap for a future problem. Probably worth extracting out an `apply_header` helper method that both `headers` and `reinit_options` could use so that it doesn't have any side effects which modify state in the ClientDSL object